httpx/models.py

```diff
@@ -660,8 +654,7 @@ def __init__(
         *,
         http_version: str = None,
         headers: HeaderTypes = None,
-        content: ResponseContent = None,
-        on_close: typing.Callable = None,
+        content: typing.Union[bytes, Stream] = None,
```

Comment:
`HTTP11Connection` and `HTTP2Connection` pass a `Stream` instance, but do we every pass raw `bytes` anywhere else in the code base? I'm thinking maybe this can be scoped down to `stream: Stream = None`.

Reply:
See also your own comment in the PR description: "Responses accept a stream argument".

Reply:
So, right now we're only ever passing `content=<bytes>` in the test cases.

We may want to have seperate `content: bytes = None` and `stream: Stream = None` arguments. Instantiating `Response` instances isn't something that users will typically do, but we probably do want it to be doable, and we probably *don't* want to force users to pass a stream instance for the simple case.

Reply:
Once we have a sync interface on the `Response` we can have a more graceful implementation here, too, along these lines...

```python
if stream is None:
    self.stream = ByteStream(content)
    self.read()
else:
    self.stream = stream
```

And drop the `_raw_content` special casing that we currently have in a few places.

Reply:
Actually we happen to be passing `content=...` in the ASGI case at the moment, but only for convience. We could easily change that around to passing a concrete stream instance.